### DISCUSSION TITLE
Where does the new project generator live?

### DISCUSSION THREAD

**Discussion by BradWalker** [apache/netbeans#4717] [General]
I'm doing some work on the Netbeans platform and have a simple question those of you out there.

Here is a picture of what I'm looking at.

![image](https://user-images.githubusercontent.com/930871/193433231-ca667d87-12bd-44cd-8361-6ddc6ca7a121.png)

I'm curious to know where in the source tree do I go looking to find the generator that creates the template for the highlighted JForm. A sample of the generator output is listed in the window just beside the highlighted text.

Thanks.

**Comment by mbien** [comment]
you mean where the form designer templates are in the repo?

that would be here:
https://github.com/apache/netbeans/tree/master/java/form/src/org/netbeans/modules/form/resources/templates

in netbeans you can find them under:
tools -> templates -> Swing GUI Forms

**Reply by BradWalker** [reply]
Hey @mbien , that was exactly the pointer that I needed! Thanks.

Let me ask one more question. Where in the source tree does the GUI designer live? The code that actually updates the source file template when GUI changes are made.

**Reply by neilcsmith-net** [reply]
@BradWalker presume you're looking for https://github.com/apache/netbeans/blob/master/java/form/src/org/netbeans/modules/form/ and in particular `JavaCodeGenerator` and `GandalfPersistenceManager`?
